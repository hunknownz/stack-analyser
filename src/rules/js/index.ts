import './angular.js';
import './bootstrap.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './hotjar.js';
import './koa.js';
import './prettier.js';
import './prisma.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
