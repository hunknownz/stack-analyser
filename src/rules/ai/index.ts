import './anthropic.js';
import './aws.bedrock.js';
import './aws.comprehend.js';
import './aws.polly.js';
import './aws.rekognition.js';
import './aws.sagemaker.js';
import './aws.transcribe.js';
import './aws.translate.js';
import './azure.openai.js';
import './cohere.js';
import './deepseek.js';
import './gcp.aiplatform.js';
import './gcp.dialogflow.js';
import './gcp.language.js';
import './gcp.speech.js';
import './gcp.translate.js';
import './gcp.vertex.js';
import './gcp.vision.js';
import './geminiai.js';
import './groq.js';
import './huggingface.js';
import './mistralai.js';
import './ollama.js';
import './openai.js';
import './perplexityai.js';
import './vercel.ai.js';
import './xai.js';
