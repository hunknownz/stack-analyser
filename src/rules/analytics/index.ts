import './amplitude.js';
import './googleanalytics.js';
import './vercel.analytics.js';
