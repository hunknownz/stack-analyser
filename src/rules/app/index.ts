import './caddy.js';
import './discourse.js';
import './grafana.js';
import './httpd.js';
import './kibana.js';
import './kong.js';
import './nginx.js';
import './traefik.js';
import './zookeeper.js';
