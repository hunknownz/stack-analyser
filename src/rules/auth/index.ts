import './auth0.js';
import './aws.cognito.js';
import './betterauth.js';
import './clerk.js';
import './frontegg.js';
import './fusionauth.js';
import './logto.js';
import './okta.js';
import './orysh.js';
import './stytch.js';
import './supabase.auth.js';
import './workos.js';
