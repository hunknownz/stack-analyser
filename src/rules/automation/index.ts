import './browerbase.js';
import './browseruse.js';
import './firecrawl.js';
import './inngest.js';
import './n8n.js';
