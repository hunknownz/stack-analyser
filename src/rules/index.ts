import './ai/index.js';
import './analytics/index.js';
import './auth/index.js';
import './app/index.js';
import './automation/index.js';
import './ci/index.js';
import './cms/index.js';
import './collaboration/index.js';
import './communication/index.js';
import './cloud/index.js';
import './crm/index.js';
import './db/index.js';
import './etl/index.js';
import './framework/index.js';
import './hosting/index.js';
import './iac/index.js';
import './iconset/index.js';
import './language/index.js';
import './linter/index.js';
import './maps/index.js';
import './monitoring/index.js';
import './notification/index.js';
import './network/index.js';
import './payment/index.js';
import './package_manager/index.js';
import './orm/index.js';
import './queue/index.js';
import './runtime/index.js';
import './saas/index.js';
import './security/index.js';
import './spec/index.js';
import './ssg/index.js';
import './storage/index.js';
import './test/index.js';
import './tool/index.js';
import './ui/index.js';
import './ui_framework/index.js';
