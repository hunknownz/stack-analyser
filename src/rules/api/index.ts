import './aws.kinesis.js';
import './aws.rekognition.js';
import './aws.sagemaker.js';
import './gcp.aiplatform.js';
import './gcp.dialogflow.js';
import './gcp.kms.js';
import './gcp.language.js';
import './gcp.maps.js';
import './gcp.speech.js';
import './gcp.translate.js';
import './gcp.vision.js';
import './sitecore.xmcloud.js';
import './vercel.ai.js';
