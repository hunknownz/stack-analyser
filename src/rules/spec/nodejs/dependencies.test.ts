import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import type { AllowedKeys } from '../../../index.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';
import { rawList } from '../../../rules.js';

const dependencies: Record<string, string> = {};
for (const item of rawList) {
  if (item.type !== 'dependency' || item.ref.type !== 'npm') {
    continue;
  }
  dependencies['example' in item.ref ? item.ref.example : item.ref.name] =
    '0.0.0';
}
const packageJson = {
  name: 'test',
  dependencies,
};

describe('npm', () => {
  it('should match everything', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['package.json'],
        },
        files: {
          '/package.json': JSON.stringify(packageJson),
        },
      }),
    });
    const match: AllowedKeys[] = [
      'algolia',
      'angular',
      'auth0',
      'aws',
      'bootstrap',
      'browserstack',
      'couchbase',
      'cypressci',
      'datadog',
      'dynamodb',
      'elasticsearch',
      'esbuild',
      'eslint',
      'express',
      'fastify',
      'gcp',
      'gcp.cloudrun',
      'gcp.gce',
      'gcp.pubsub',
      'github',
      'hotjar',
      'influxdb',
      'koa',
      'logrocket',
      'mailjet',
      'mariadb',
      'memcached',
      'mongodb',
      'mysql',
      'neo4j',
      'netlify',
      'newrelic',
      'nodejs',
      'postgresql',
      'prettier',
      'prisma',
      'rabbitmq',
      'react',
      'reactemail',
      'redis',
      'relativeci',
      'renovate',
      'rollup',
      'sentry',
      'sequelize',
      'snyk',
      'socketio',
      'sqreen',
      'storybook',
      'strapi',
      'tailwind',
      'typescript',
      'vercel',
      'vite',
      'vue',
      'webpack',
    ];
    expect(
      Array.from(flatten(res, { merge: true }).techs).sort()
    ).toStrictEqual(match);
  });

  it('should match nothing', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['package.json'],
        },
        files: {
          '/package.json': JSON.stringify({ foo: 'bar' }),
        },
      }),
    });
    const match: AllowedKeys[] = ['nodejs'];
    expect(
      Array.from(flatten(res, { merge: true }).techs).sort()
    ).toStrictEqual(match);
  });
});
