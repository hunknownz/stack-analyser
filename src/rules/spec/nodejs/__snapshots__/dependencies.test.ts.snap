// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`npm > should match everything 1`] = `
[
  "algolia",
  "angular",
  "auth0",
  "aws",
  "bootstrap",
  "browserstack",
  "couchbase",
  "cypressci",
  "datadog",
  "dynamodb",
  "elasticsearch",
  "esbuild",
  "eslint",
  "express",
  "fastify",
  "gcp",
  "gcp.cloudrun",
  "gcp.gce",
  "gcp.gcs",
  "gcp.pubsub",
  "github",
  "hotjar",
  "influxdb",
  "koa",
  "logrocket",
  "mailjet",
  "mariadb",
  "memcached",
  "mongodb",
  "mysql",
  "neo4j",
  "netlify",
  "newrelic",
  "nodejs",
  "postgresql",
  "prettier",
  "prisma",
  "rabbitmq",
  "react",
  "reactemail",
  "redis",
  "relativeci",
  "renovate",
  "rollup",
  "sentry",
  "sequelize",
  "snyk",
  "socketio",
  "sqreen",
  "storybook",
  "strapi",
  "tailwind",
  "typescript",
  "vercel",
  "vite",
  "vue",
  "webpack",
]
`;
